source/develop/release-management/features/sla/soft-host-to-vm-affinity-support.html.md

```diff
@@ -0,0 +1,110 @@
+---
+title: soft-host-to-vm-affinity-support
+category: feature
+authors: yquinn,msivak
+wiki_category: Feature
+wiki_title: Features/soft-host-to-vm-affinity-support
+wiki_last_updated: 2016-11-23
+---
+
+# Soft host to VM affinity support and Rack use case
+
+## Summary
+Certain set of VMs form a logical management unit and should run on a certain set of hosts for SLA or management (e.g. a separate rack for each customer). The VMs can run anywhere in case the dedicated rack needs to be turned off, but should return to their dedicated hosts once the rack is back up.
+
+This would be best implemented by enhancing the current VM to VM soft affinity rules to support hosts as well. The other piece that is needed is to add affinity based balancing operation to allow the VMs to return to the proper host when it becomes available.
+
+### Owners
+
+*   Name: Yanir Quinn (yquinn)
+*   Email: <yquinn@redhat.com>
+    
+## Detailed Description
+ 
+Add additional affinity lists and relationships to the existing affinity object so it would contain:
+
+*   Vms_affinity_enabled flag (true/false)
+*   VM list
+*   VM to VM affinity rule (positive weak, pos. strong, negative weak, neg. strong,disabled)
+*   Host list
+*   VM to Host affinity rule (positive weak, pos. strong, negative weak, neg. strong)
+*   Any other additional resource
+*   Rule for the additional resource (positive weak, pos. strong, negative weak, neg. strong)
+
+### Data Base
+
+the plan is to add more resources like hosts to affinity groups ,
+The table structure of affinity_group_members will be altered
+From :  affinity_group_id (foreign key to affinity_groups + delete cascade) |  vm_id (foreign key to vm_static + delete cascade)
+To :      affinity_group_id (foreign key to affinity_groups + delete cascade) | vm_id (foreign key to vm_static + delete cascade) - change modifier to enable null   | vds_id (foreign key to vds_static + delete cascade) | (other resource ids in the future...)
+(any additional future resource can be added here with an index and a foreign key)
+
+* Added a constraint to check that at least one column out of vm_id and vds_id is not null for each entry.
+
+
+The table structure of affinity_groups will altered by adding a new none column:
+affinity_groups: id, name, cluster_id (foreign key to vds_groups + delete cascade), positive -> vm_positive , enforcing -> vm_enforcing ,vds_positive,vds_enforcing ,vms_affinity_enabled (true/false  - default false) 
+
+NOTE : The additional “vms_affinity_enabled”  flag is needed for affinity group to be able to express that the group of its VMs have no relationship to each other.
```

Comment:
Just a grammar issue: "group of its VMs have no..." should probably just say "VMs in an affinity group have no..".